src/com/maddyhome/idea/vim/extension/highlightedyank/VimHighlightedYank.kt

```diff
@@ -0,0 +1,184 @@
+ *
+ * let g:highlightedyank_highlight_duration = "1000"
+
+ * A negative number makes the highlight persistent.
+ * let g:highlightedyank_highlight_duration = "-1"
+
+ * if you want to change background color of highlight you can provide the rgba of the color you want e.g.
+ * let g:highlightedyank_highlight_color = "rgba(160, 160, 160, 155)"
+ *
+ * When a new text is yanked or user starts editing, the old highlighting would be deleted.
+ */
+class VimHighlightedYank: VimExtension {
+  override fun getName() = "highlightedyank"
+
+  override fun init() {
+    highlightEnabled = true
+  }
+
+
+  override fun dispose() {
+    super.dispose()
+    highlightEnabled = false
+  }
+
+  companion object {
+    var highlightEnabled: Boolean = false
+    private val highlightHandler = HighlightHandler()
+
+    fun highlightYankRange(editor: Editor, range: TextRange) {
+      highlightHandler.highlightYankRange(editor, range, highlightEnabled)
+    }
+
+    fun clearAllYankHighlighters() {
+      highlightHandler.clearAllYankHighlighters()
+    }
+  }
+
+  private class HighlightHandler {
+    private val yankHighlighters: MutableSet<Pair<Editor, RangeHighlighter>> = mutableSetOf()
+
+    fun highlightYankRange(editor: Editor, range: TextRange, highlightEnabled: Boolean) {
+      if(!highlightEnabled) return
+
+      //from vim-highlightedyank docs: When a new text is yanked or user starts editing, the old highlighting would be deleted
+      clearAllYankHighlighters()
+
+      if (range.isMultiple) {
+        for (i in 0 until range.size()) {
+          highlightSingleRange(editor, range.startOffsets[i]..range.endOffsets[i])
+        }
+      } else {
+        highlightSingleRange(editor, range.startOffset..range.endOffset)
+      }
+    }
+
+    fun clearAllYankHighlighters() {
+      yankHighlighters.forEach { (editor, highlighter) ->
+          editor.markupModel.removeHighlighter(highlighter)
+      }
+
+      yankHighlighters.clear()
+    }
+
+    private fun highlightSingleRange(editor: Editor, range: ClosedRange<Int>) {
+      val textAttributes = TextAttributes(
+        null,
+        extractUsersHighlightColor(),
+        editor.colorsScheme.getColor(EditorColors.CARET_COLOR),
+        EffectType.SEARCH_MATCH, Font.PLAIN
+      )
+
+      val highlighter = editor.markupModel.addRangeHighlighter(
+        range.start,
+        range.endInclusive,
+        HighlighterLayer.SELECTION - 1,
+        textAttributes,
+        HighlighterTargetArea.EXACT_RANGE
+      )
+
+      yankHighlighters.add(Pair(editor, highlighter))
+
+      setClearHighlightRangeTimer(editor, highlighter)
+    }
+
+    private fun setClearHighlightRangeTimer(editor: Editor, highlighter: RangeHighlighter) {
+      val timeout = extractUsersHighlightDuration()
+
+      //from vim-highlightedyank docs: A negative number makes the highlight persistent.
+      if(timeout >= 0) {
+        Executors.newSingleThreadScheduledExecutor().schedule({
```

Comment:
I suggest to use `pooledThreadSingleAlarm` from `SingleAlarm.kt`.

Reply:
Hmmm... this is weird because it doesn't really work at all with `pooledThreadSingleAlarm`. I've tried it with many different options and the function inside `pooledThreadSingleAlarm` its just not called at all

Reply:
Okay, let's don't use it :)